data/blog/intro-to-entity-framework-core-in-net-6.md

```diff
@@ -0,0 +1,586 @@
+        if (!isEmpty)
+        {
+            dbContext.Products.AddRange(
+                new() { Name = "Laptop", Price = 1000, Quantity = 50 },
+                new() { Name = "Mouse", Price = 50, Quantity = 25 },
+                new() { Name = "Keyboard", Price = 100, Quantity = 10 }
+            );
+
+            dbContext.SaveChanges();
+        }
+    }
+}
+```
+
+### Display.cs
+
+```cs
+using Ef_Intro.Data;
+using Ef_Intro.Entities;
+
+namespace Ef_Intro.Utilities;
+
+public static class Display
+{
+    public static void Print(string message)
+    {
+        Console.WriteLine(message);
+    }
+
+    public static void Products(IEnumerable<Product> products)
+    {
+        foreach (var product in products)
+        {
+            Console.WriteLine("Id: " + product.Id);
+            Console.WriteLine("Name: " + product.Name);
+            Console.WriteLine("Price: " + product.Price);
+            Console.WriteLine("Quantity: " + product.Quantity);
+            Console.WriteLine("===============================");
+        }
+    }
+
+    public static int Menu()
+    {
+        Console.WriteLine("1. Print products");
+        Console.WriteLine("2. Add product");
+        Console.WriteLine("3. Update product price");
+        Console.WriteLine("4. Delete product");
+        Console.WriteLine("5. Exit");
+        Console.Write("Enter your choice: ");
+        if (int.TryParse(Console.ReadLine(), out int choice))
+        {
+            return choice;
+        }
+
+        return -1;
+    }
+
+    public static void ResetConsole()
+    {
+        Console.WriteLine("\nPress any key to continue . . .");
+        Console.ReadKey();
+        Console.Clear();
+    }
+
+    internal static void InvalidInput()
+    {
+        Console.WriteLine("The input you submitted is not in a valid format");
+        Console.WriteLine("\nPress any key to continue . . .");
+        Console.ReadKey();
+        Console.Clear();
+    }
+}
+```
+
+The entry point finally looks like.
+
+### `Program.cs`
+
+```cs
+using Ef_Intro.Data;
+using Ef_Intro.Services;
+using Ef_Intro.Utilities;
+
+internal class Program
+{
+    private static void Main()
+    {
+        bool exit = false;
+        // We instantiate our db context
+        using var dbContext = new AppDbContext();
+        IProductsService productsService = new ProductsService(dbContext);
+        IInputManager inputManager = new InputManager(productsService);
+
+        // We initialize the database
+        Database.InitDatabase(dbContext);
+
+        while (!exit)
+        {
+            int input = Display.Menu();
+            exit = !inputManager.Proceed(input);
+            Display.ResetConsole();
+        }
+
+        // We close the db context
+        dbContext.Dispose();
+    }
+}
+```
+
+And by running the command
+
+```shell
+dotnet run
+```
+
+The output should be
+
+![Output result](/static/images/posts/intro-to-entity-framework-core-in-net-6/output.png)
```

Comment:
Can you put this output as a code block?

Reply:
Okay I will